src/compiler/checker.ts

```diff
@@ -36423,8 +36440,8 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             case SyntaxKind.LessThanEqualsToken:
             case SyntaxKind.GreaterThanEqualsToken:
                 if (checkForDisallowedESSymbolOperand(operator)) {
-                    leftType = getBaseTypeOfLiteralType(checkNonNullType(leftType, left));
-                    rightType = getBaseTypeOfLiteralType(checkNonNullType(rightType, right));
+                    leftType = getBaseTypeOfLiteralTypeForComparison(checkNonNullType(leftType, left));

```

Comment:
I suspect this isn't the only case; the `AmpersandAmpersandToken` block below also uses the old function. But, changing that produces an... odd result:

```
>rf6 : 0 | E.b | E.c
>a6 && a6 : 0 | E.b | E.c
```